roles/gpu_operator_bundle_from_commit/files/operator_image_builder_script.yml

```diff
@@ -22,21 +20,31 @@ data:
 
     git show --quiet
 
-    if [ "${BUILDER_FROM_IMAGE}" ]; then
-      # if provided, use custom 'FROM ... as builder' image
-      sed "s|FROM golang:.* as builder|FROM ${BUILDER_FROM_IMAGE} as builder|" -i ${CONTAINER_FILE}
-    fi
-
-    # avoid docker.io quotas ...
-    sed -i 's|FROM nvidia/cuda:|FROM nvcr.io/nvidia/cuda:|' ${CONTAINER_FILE}
-
     cat <<EOF > /usr/local/bin/docker
     #! /bin/sh
     exec podman --cgroup-manager=cgroupfs "\$@"
     EOF
     chmod u+x /usr/local/bin/docker
 
-    make docker-build IMG=${OPERATOR_IMAGE_NAME}
+    CONTAINER_FILE=./Dockerfile
+    if [ -f ${CONTAINER_FILE} ]; then
+      # TODO: Remove this backwards compatible code once we stop testing versions that do not use the new Makefile
+      if [ "${BUILDER_FROM_IMAGE}" ]; then
+        # if provided, use custom 'FROM ... as builder' image
+        sed "s|FROM golang:.* as builder|FROM ${BUILDER_FROM_IMAGE} as builder|" -i ${CONTAINER_FILE}
+      fi
+
+      # avoid docker.io quotas ...
+      sed -i 's|FROM nvidia/cuda:|FROM nvcr.io/nvidia/cuda:|' ${CONTAINER_FILE}
+      make docker-build IMG=${OPERATOR_IMAGE_NAME}
+    else
+      # if provided, use custom 'FROM ... as builder' image by setting BUILDER_IMAGE variable
+      # avoid docker.io quotas by setting CUDA_IMAGE
+      # This generates an images OUT_IMAGE (based on OPERATOR_IMAGE_NAME)
+      make docker-image \
+        REGISTRY="" \
+        OUT_IMAGE=${OPERATOR_IMAGE_NAME} CUDA_IMAGE=nvcr.io/nvidia/cuda ${BUILDER_FROM_IMAGE:+BUILDER_IMAGE=${BUILDER_FROM_IMAGE}}
```

Comment:
could you align `CUDA_IMAGE=...` and `${BUILDER_FROM_IMAGE}...` with `REGISTRY=""` ?

Reply:
Will do. I had some concern over adding `${BUILDER_FROM_IMAGE:+BUILDER_IMAGE=${BUILDER_FROM_IMAGE}}` as the last line as this could be empty and I was not 100% certain how bash would handle that as a continuation line. A quick reorder should address this.